contract/src/pegasus.js

```diff
@@ -164,22 +164,26 @@ const makeCourier = ({
 
   /** @type {Sender} */
   const send = async (zcfSeat, depositAddress) => {
-    const amount = await zcfSeat.getAmountAllocated('Transfer', localBrand);
-    const packet = localAmountToPacket(amount, depositAddress);
-
-    // Retain the payment.  We must not proceed on failure.
-    retain(zcfSeat, { Transfer: amount });
-
-    // The payment is already escrowed, and proposed to retain, so try sending.
-    return sendTransferPacket(connection, packet).then(
-      _ => zcfSeat.exit(),
-      reason => {
-        // Return the payment to the seat, if possible.
-        redeem(zcfSeat, { Transfer: amount });
-        // Reflect the error back to the seat.
-        zcfSeat.kickOut(reason);
-      },
-    );
+    const tryToSend = async () => {
+      const amount = zcfSeat.getAmountAllocated('Transfer', localBrand);
+      const packet = localAmountToPacket(amount, depositAddress);
+
+      // Retain the payment.  We must not proceed on failure.
+      retain(zcfSeat, { Transfer: amount });
+
+      // The payment is already escrowed, and proposed to retain, so try sending.
+      return sendTransferPacket(connection, packet).then(
+        _ => zcfSeat.exit(),
+        reason => {
+          // Return the payment to the seat, if possible.
+          redeem(zcfSeat, { Transfer: amount });
+          throw reason;
+        },
+      );
+    };
+
+    // Reflect any error back to the seat.
+    return tryToSend().catch(reason => zcfSeat.kickOut(reason));
```

Comment:
IIRC, `kickOut` itself throws, so case this will cause `send` to return an eventually rejected promise with no code in here to handle that rejection. This is or is not ok depending on how `send` is used.

Reply:
I checked, and `kickOut` does not throw.  It would have surprised me if it did.

Reply:
Caught by your type system!

`kickOut` now returns the reason, so in the error case this code now returns a promise for an error. I gather that your intention is that `send` returns a promise for void that, in this circumstance, fulfills rather than rejects, so just put curlies around the body.